src/ixdat/plotters/ms_plotter.py

```diff
@@ -242,14 +316,29 @@ def plot_vs(
             ax,
             axes,
             measurement,
+            use_quantity,
         )
         ax = specs_this_axis["ax"]
         v_list = specs_this_axis["v_list"]
         tspan_bg = specs_this_axis["tspan_bg"]
         unit = specs_this_axis["unit"]
-        unit_factor = specs_this_axis["unit_factor"]
+        # unit_factor = specs_this_axis["unit_factor"]
+
+        t, x = measurement.grab(
+            x_name, tspan=tspan, include_endpoints=True, return_quantity=use_quantity
+        )
+
+        if use_quantity and x_unit and not x.check(x_unit):
+            error_message = (
```

Comment:
see previous comment on this. 
Also is there a way to avoid repeating such similar code?